src/Float.mo

```diff
@@ -85,62 +406,376 @@ module {
+  /// less(x, nan)     => false for any Float x
+  /// ```
+  ///
+  /// Example:
+  /// ```motoko
+  /// import Float "mo:base/Float";
+  ///
+  /// Float.less(Float.e, Float.pi) // => true
+  /// ```
   public func less(x : Float, y : Float) : Bool { x < y };
 
   /// Returns `x <= y`.
+  ///
+  /// Special cases:
+  /// ```
+  /// lessOrEqual(+0.0, -0.0) => true
+  /// lessOrEqual(-0.0, +0.0) => true
+  /// lessOrEqual(nan, y)     => false for any Float y
+  /// lessOrEqual(x, nan)     => false for any Float x
+  /// ```
+  ///
+  /// Example:
+  /// ```motoko
+  /// import Float "mo:base/Float";
+  ///
+  /// Float.lessOrEqual(0.123, 0.1234) // => true
+  /// ```
   public func lessOrEqual(x : Float, y : Float) : Bool { x <= y };
 
   /// Returns `x > y`.
+  ///
+  /// Special cases:
+  /// ```
+  /// greater(+0.0, -0.0) => false
+  /// greater(-0.0, +0.0) => false
+  /// greater(nan, y)     => false for any Float y
+  /// greater(x, nan)     => false for any Float x
+  /// ```
+  ///
+  /// Example:
+  /// ```motoko
+  /// import Float "mo:base/Float";
+  ///
+  /// Float.greater(Float.pi, Float.e) // => true
+  /// ```
   public func greater(x : Float, y : Float) : Bool { x > y };
 
   /// Returns `x >= y`.
+  ///
+  /// Special cases:
+  /// ```
+  /// greaterOrEqual(+0.0, -0.0) => true
+  /// greaterOrEqual(-0.0, +0.0) => true
+  /// greaterOrEqual(nan, y)     => false for any Float y
+  /// greaterOrEqual(x, nan)     => false for any Float x
+  /// ```
+  ///
+  /// Example:
+  /// ```motoko
+  /// import Float "mo:base/Float";
+  ///
+  /// Float.greaterOrEqual(0.1234, 0.123) // => true
+  /// ```
   public func greaterOrEqual(x : Float, y : Float) : Bool { x >= y };
 
   /// Returns the order of `x` and `y`.
+  ///
+  /// Note: This operation is discouraged as it does not consider numerical errors for equality, see comment above.
+  ///
+  /// Issue: Undefined behavior for `nan`, not defining a total number order.
+  ///
+  /// Special cases:
+  /// ```
+  /// compare(+0.0, -0.0) => #equal
+  /// compare(-0.0, +0.0) => #equal
+  /// compare(nan, y) is undefined for any Float y
+  /// compare(x, nan) is undefined for any Float x
+  /// ```
+  ///
+  /// Example:
+  /// ```motoko
+  /// import Float "mo:base/Float";
+  ///
+  /// Float.compare(0.123, 0.1234) // => #less
+  /// ```
   public func compare(x : Float, y : Float) : { #less; #equal; #greater } {
     if (x < y) { #less } else if (x == y) { #equal } else { #greater }
   };
 
   /// Returns the negation of `x`, `-x` .
-  public func neq(x : Float) : Float { -x };
+  ///
+  /// Changes the sign bit for infinity.
+  /// Issue: Inconsistent behavior for zero and `NaN`. Probably related to
+  /// https://github.com/dfinity/motoko/issues/3646
+  ///
+  /// Example:
+  /// ```motoko
+  /// import Float "mo:base/Float";
+  ///
+  /// Float.neq(1.23) // => -1.23
+  /// ```
+  public func neq(x : Float) : Float { -x }; // Typo: Should be changed to `neg`
```

Comment:
indeed. I guess we could deprecate `neq` and add `neg` (but also in a separate PR)

Reply:
I will prepare that in a separate PR. Good idea to deprecate the misspelled function.